library/ssl_tls13_client.c

```diff
@@ -1957,16 +1957,16 @@ static int ssl_client_hello_write( mbedtls_ssl_context* ssl,
         total_ext_len += cur_ext_len;
         buf += cur_ext_len;
 
-        if( ret == 0 ) ssl->handshake->extensions_present |= KEY_SHARE_EXTENSION;
+        if( ret == 0 ) mbedtls_ssl_extensions_present( ssl, KEY_SHARE_EXTENSION, 1 );
     }
-    else if( ssl->handshake->extensions_present & SUPPORTED_GROUPS_EXTENSION && ssl->handshake->extensions_present & SIGNATURE_ALGORITHM_EXTENSION )
+    else if( mbedtls_ssl_extensions_present( ssl, SUPPORTED_GROUPS_EXTENSION | SIGNATURE_ALGORITHM_EXTENSION, 0 ) )
```

Comment:
Blocker: This is functionally incorrect since the present implementaiton of `mbedtls_ssl_extensions_present()` only checks whether a non-zero part of the flag is present, not whether the entire flag is present.

Reply:
Fixed `mbedtls_ssl_extensions_present()` to support a conjunction of extensions

Reply:
It may be easier to independently call `mbedtls_ssl_extensions_present()` for each extension rather than calling it once to check several extensions though...